src/OrbitGl/TrackContainer.cpp

```diff
@@ -417,15 +415,29 @@ void TrackContainer::DrawThreadDependencyArrow(PrimitiveAssembler& primitive_ass
       Vec2{x, start_arrow_y}, Vec2{layout_->GetThreadDependencyArrowBodyWidth(), arrow_body_height},
       Vec2{layout_->GetThreadDependencyArrowHeadWidth(),
            layout_->GetThreadDependencyArrowHeadHeight()},
-      GlCanvas::kZValueOverlay, kArrowColor, arrow_direction);
+      GlCanvas::kZValueOverlay, arrow_color, arrow_direction);
+}
+
+void TrackContainer::DrawThreadDependency(PrimitiveAssembler& primitive_assembler,
+                                          PickingMode picking_mode) {
+  if (app_->selected_thread_state_slice() != std::nullopt) {
+    const Color kSelectedSliceArrowColor{255, 255, 255, 255};
+    DrawThreadDependencyArrow(app_->selected_thread_state_slice().value(), kSelectedSliceArrowColor,
+                              primitive_assembler, picking_mode);
+  }
+  if (app_->hovered_thread_state_slice() != std::nullopt) {
+    const Color kHoveredSliceArrowColor{255, 255, 255, 64};
+    DrawThreadDependencyArrow(app_->hovered_thread_state_slice().value(), kHoveredSliceArrowColor,
+                              primitive_assembler, picking_mode);
+  }
 }
 
 void TrackContainer::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer& text_renderer,
                             const DrawContext& draw_context) {
   CaptureViewElement::DoDraw(primitive_assembler, text_renderer, draw_context);
 
   DrawIncompleteDataIntervals(primitive_assembler, draw_context.picking_mode);
-  DrawThreadDependencyArrow(primitive_assembler, draw_context.picking_mode);
+  DrawThreadDependency(primitive_assembler, draw_context.picking_mode);
```

Comment:
Nit: I would make `if (picking_mode != PickingMode::kNone) { ` here and save pasing the parameter to the method.

Reply:
I decided not to do this for consistency since the other draw functions check for picking mode inside the function.